File: algorithms/2022.convert-1d-array-into-2d-array/solution.rb
# @param {Integer[]} original
# @param {Integer} m
# @param {Integer} n
# @return {Integer[][]}
def construct2_d_array(original, m, n)
  if original.size == m * n
    original.each_slice(n).to_a
  else
    []
  end
end

File: algorithms/2022.convert-1d-array-into-2d-array/solution.rs
impl Solution {
    pub fn construct2_d_array(original: Vec<i32>, m: i32, n: i32) -> Vec<Vec<i32>> {
        if original.len() as i32 == m * n {
            original.chunks(n as usize).map(|v| v.to_vec()).collect()
        } else {
            vec![]
        }
    }
}
